Gradle multimodule project - production build fails due to static assets missing

### Description of the bug

Multimodule project - Subproject 1 + Subproject 2

Subproject 2 uses static assets from Submodule 1

./gradlew bootRun or ./gradlew bootJar works fine but when setting "-Pvaadin.productionMode=true" frontend compilation fails:

```
* What went wrong:
Execution failed for task ':theme:vaadinBuildFrontend'.
> Webpack process exited with non-zero exit code.

(...)

Module not found: Error: Can't resolve './styles/colors.css' in ' (--truncated--) Vaadin/project/theme/build/frontend'
```
This happens for every module in the resources dir.

### Expected behavior

Production and dev builds should both complete but only dev works.

### Minimal reproducible example

Subproject1 - Theme

build.gradle:

```
plugins {
    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
    id 'java'
    id 'com.vaadin'
}

group 'com.xxx.designsystem.vaadin'
version '1.0-SNAPSHOT'

sourceCompatibility = JavaVersion.VERSION_11
targetCompatibility = JavaVersion.VERSION_11

repositories {
    mavenCentral()
    maven { setUrl("https://maven.vaadin.com/vaadin-prereleases") }
    maven { setUrl("https://maven.vaadin.com/vaadin-addons") }
}

dependencyManagement {
    imports {
        mavenBom "com.vaadin:vaadin-bom:$vaadinVersion"
    }
}

dependencies {
    implementation 'com.vaadin.componentfactory:tooltip:2.0.2'
    implementation 'org.vaadin.gatanaso:multiselect-combo-box-flow:4.0.0-rc2'
}

test {
    useJUnitPlatform()
}
```

Subproject2 - Vaadin demo app

build.gradle:

```
plugins {
    id 'org.springframework.boot' version '2.5.12'
    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
    id 'java'
    id 'com.vaadin'
}

group 'com.xxx.designsystem.demo'
version '1.0-SNAPSHOT'

sourceCompatibility = JavaVersion.VERSION_11
targetCompatibility = JavaVersion.VERSION_11

repositories {
    mavenCentral()
    maven { setUrl("https://maven.vaadin.com/vaadin-prereleases") }
    maven { setUrl("https://maven.vaadin.com/vaadin-addons") }
}

configurations {
    developmentOnly
    runtimeClasspath {
        extendsFrom developmentOnly
    }
}

dependencyManagement {
    imports {
        mavenBom "com.vaadin:vaadin-bom:$vaadinVersion"
    }
}

dependencies {
    implementation project(":theme")
    implementation('com.vaadin:vaadin-spring-boot-starter')
    implementation 'com.vaadin.componentfactory:tooltip:2.0.2'
    implementation 'org.vaadin.gatanaso:multiselect-combo-box-flow:4.0.0-rc2'
    developmentOnly 'org.springframework.boot:spring-boot-devtools'
    testImplementation('org.springframework.boot:spring-boot-starter-test') {
        exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
    }
}

test {
    useJUnitPlatform()
}
```

Root project - build.gradle:
```
plugins {
    id 'java'
}

group 'com.xxx.designsystem'
version '1.0-SNAPSHOT'

repositories {
    mavenCentral()
    maven { setUrl("https://maven.vaadin.com/vaadin-prereleases") }
    maven { setUrl("https://maven.vaadin.com/vaadin-addons") }
}

dependencies {
    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.0'
    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
}

test {
    useJUnitPlatform()
}
```

Add static resources to theme/src/main/resources/META-INF/resources/whatever.css and build a component that uses it.
Then use that component in vaadin-demo-app (subproject2)

```
./gradlew bootRun -> works
./gradlew bootJar -> works
./gradlew "-Pvaadin.productionMode=true" bootRun -> fails
./gradlew "-Pvaadin.productionMode=true" bootRun -> fails
```

### Versions

Vaadin 23.1.6
'io.spring.dependency-management' version '1.0.11.RELEASE'
gradle: 6.8.3
'org.springframework.boot' version '2.5.12'

--- a/flow-plugins/flow-gradle-plugin/src/functionalTest/kotlin/com/vaadin/gradle/VaadinSmokeTest.kt
+++ b/flow-plugins/flow-gradle-plugin/src/functionalTest/kotlin/com/vaadin/gradle/VaadinSmokeTest.kt
@@ -299,4 +299,47 @@ class VaadinSmokeTest : AbstractGradleTest() {
             File(testProject.dir, "src/main/frontend/generated/index.ts").exists()
         }
     }
+
+    /**
+     * Tests that build works with resources from classpath, not only from
+     * frontend directory
+     *
+     * https://github.com/vaadin/flow/issues/14420
+     */
+    @Test
+    fun vaadinBuildFrontendShouldScanForResourcesOnClasspath() {
+        testProject.newFile("src/main/java/org/vaadin/example/MainView.java", """
+            package org.vaadin.example;
+
+            import com.vaadin.flow.component.dependency.CssImport;
+            import com.vaadin.flow.component.html.Div;
+            import com.vaadin.flow.component.html.Span;
+            import com.vaadin.flow.router.Route;
+
+            @Route("")
+            @CssImport("./mystyle.css")
+            public class MainView extends Div {
+
+                public MainView() {
+                    add(new Span("It works!"));
+                }
+            }
+        """.trimIndent())
+        testProject.newFile("src/main/resources/META-INF/resources/frontend/mystyle.css", """
+            #testme {
+                background-color: red;
+                width: 100px;
+                height: 100px;
+                display: block;
+            }
+        """.trimIndent())
+
+        val result: BuildResult = testProject.build("-Pvaadin.productionMode", "build")
+        result.expectTaskSucceded("vaadinPrepareFrontend")
+        result.expectTaskSucceded("vaadinBuildFrontend")
+
+        val cssFile = File(testProject.dir, "build/flow-frontend/mystyle.css")
+        expect(true, cssFile.toString()) { cssFile.exists() }
+
+    }
 }

--- a/flow-plugins/flow-gradle-plugin/src/functionalTest/kotlin/com/vaadin/gradle/VaadinSmokeTest.kt
+++ b/flow-plugins/flow-gradle-plugin/src/functionalTest/kotlin/com/vaadin/gradle/VaadinSmokeTest.kt
@@ -312,12 +312,14 @@ class VaadinSmokeTest : AbstractGradleTest() {
             package org.vaadin.example;
 
             import com.vaadin.flow.component.dependency.CssImport;
+            import com.vaadin.flow.component.dependency.NpmPackage;
             import com.vaadin.flow.component.html.Div;
             import com.vaadin.flow.component.html.Span;
             import com.vaadin.flow.router.Route;
 
             @Route("")
             @CssImport("./mystyle.css")
+            @NpmPackage(value = "@vaadin/vaadin-themable-mixin", version = "23.3.0-alpha1")
             public class MainView extends Div {
 
                 public MainView() {

--- a/flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/GradlePluginAdapter.kt
+++ b/flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/GradlePluginAdapter.kt
@@ -53,6 +53,9 @@ internal class GradlePluginAdapter(val project: Project, private val isBeforePro
             .toList()
             .filter { it.exists() }
 
+        val resourcesDir: List<File> = listOfNotNull(sourceSet.getByName("main").output.resourcesDir)
+                .filter { it.exists() }
+
         // for Spring Boot project there is no "providedCompile" scope: the WAR plugin brings that in.
         val providedDeps: Configuration? = project.configurations.findByName("providedCompile")
         val servletJar: List<File> = providedDeps
@@ -60,7 +63,7 @@ internal class GradlePluginAdapter(val project: Project, private val isBeforePro
             ?.toList()
             ?: listOf()
 
-        val apis: Set<File> = (runtimeClasspathJars + classesDirs + servletJar).toSet()
+        val apis: Set<File> = (runtimeClasspathJars + classesDirs + resourcesDir + servletJar).toSet()
 
         // eagerly check that all the files/folders exist, to avoid spamming the console later on
         // see https://github.com/vaadin/vaadin-gradle-plugin/issues/38 for more details

